src/main/java/SpringWeatherForecast/WeatherController.java

```diff
@@ -0,0 +1,38 @@
+package SpringWeatherForecast;
+
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class WeatherController {
+    private final WeatherRepository repository;
+
+    DataSource source = new DataSource();
+    WeatherForecaster forecaster = new WeatherForecaster(source);
+
+
+    public WeatherController(WeatherRepository repository){
+        this.repository = repository;
+    }
+
+    @PostMapping("/api")
+    void addWeather(@RequestBody ConsolidatedWeather weather) {
```

Comment:
This methods allows a user to call an endpoint and add a weather right to the database. Are you sure this is the desired behaviour? We agreed that our API allows only to query (read) the weather, do not store it.